app/subscriber/src/features/my-reports/edit/content/ContentEditForm.tsx

```diff
@@ -60,18 +60,47 @@ export const ContentEditForm = React.forwardRef<HTMLDivElement | null, IContentE
       return err;
     };
 
+    const isContentUpdated = React.useCallback(
+      async (userContent: IContentModel, updatedContent: IContentModel) => {
+        if (
+          updatedContent.version &&
+          userContent.version &&
+          updatedContent.version > userContent.version
+        ) {
+          const key: number = userId;
+          const userVersions = userContent.versions[userId];
+          const newVersions = userContent.versions;
+          newVersions[key] = userVersions;
+
+          return { ...updatedContent, versions: newVersions };
+        } else {
+          return userContent;
+        }
+      },
+      [userId],
+    );
+
     const handleAddUpdateContent = React.useCallback(
       async (values: IReportForm, row: IReportInstanceContentForm) => {
         try {
           setSubmitting(true);
           const content = row.content;
-          if (!content) return null;
+          if (!content || content === undefined) return null;
+          const originalId = content.id;
           const err = validate(content);
           if (err.hasErrors) return null;
-          const originalId = content.id;
-          const contentResult = !content.id
-            ? await addContent(content)
-            : await updateContent(content);
+          // gets the edited content to check if the version was changed by another user
+          const updatedContent = await getContent(originalId);
```

Comment:
Making a request to get latest here isn't ideal as it's an extra request which isn't required most of the time.  Also it doesn't actually stop the issue from occurring as the content could be updated in the time it takes to send another request.  Instead attempt to save first, if the failure is for concurrency, then get the latest and merge.